src/component/CouponRow.js

```diff
@@ -0,0 +1,169 @@
+import React, {Component} from 'react';
+import {Text, View, Image, StyleSheet} from 'react-native';
+import PropTypes from 'prop-types';
+import * as Constants from '../common/constants';
+
+export default class CouponRow extends Component {
+    static propTypes = {
+        category: PropTypes.string.isRequired,
+        description: PropTypes.string.isRequired,
+        merchant: PropTypes.string,
+        used: PropTypes.bool,
+        price: PropTypes.string.isRequired,
+    };
+
+    render() {
+        let merchant = null;
+        let icon = null;
+
+        switch (this.props.category) {
```

Comment:
Instead of `switch`, I think you can create a Map of <Category, icon>.
Something like:
```
const icons = {
   'Cinema' : require();
}
```

And use it as `icons[this.props.category]` directly.

Reply:
Thinking to the future, probably the single coupon will have a categoryId, not the full name.

I think we can move the map inside a `common/categories.js`, with name and icon (still using the name as key for now) file and use it where we need.